### PR TITLE
[BUGFIX beta] Ensure that `this._super` is called by Components.

### PR DESCRIPTION
**Pull request by rwjblue**
When implementing `init` in a component subclass you must call `this._super(...arguments)`.  If you do not you will get an error.  Unfortunately, that error is **very** hard to understand and is completely unrelated to code that you have written.

This change adds a small flag (via symbol to avoid exposing this publicly) that we can use to ensure that `_super` was called properly.

---

After these changes, given the following:

``` javascript
import Ember from 'ember';

export default Ember.Component.extend({
  init() {
    this.doThings();
  }
});
```

Will throw an error that explains that you must call `_super`:

```
You must call `this._super(...arguments);` when implementing `init`
in a component. Please update ${this} to call `this._super` from `init`.
```

/cc @ef4

### CODE REVIEW

File: packages/ember-views/lib/mixins/view_support.js [hunk 1]
```diff
@@ -7,6 +7,9 @@ import { guidFor } from 'ember-metal/utils';
 import { computed } from 'ember-metal/computed';
 import { Mixin } from 'ember-metal/mixin';
 import { POST_INIT } from 'ember-runtime/system/core_object';
+import { symbol } from 'ember-metal/utils';
+
+let INIT_WAS_CALLED = symbol('INIT_WAS_CALLED');
```
Comment (ef4) [inline packages/ember-views/lib/mixins/view_support.js]: Opportunity for `const`.

Comment (rwjblue) [inline packages/ember-views/lib/mixins/view_support.js]: Good point, updated!

File: packages/ember-views/lib/mixins/view_support.js [hunk 2]
```diff
@@ -610,6 +613,7 @@ export default Mixin.create({
     this.scheduledRevalidation = false;
```
Comment (stefanpenner) [inline packages/ember-views/lib/mixins/view_support.js]: although not part of this commit, should this assignment to `this` be before the `init` ?

**Comment by stefanpenner** [comment]
LGTM

File: packages/ember-views/lib/mixins/view_support.js (hunk 2 above)
Comment (rwjblue) [inline packages/ember-views/lib/mixins/view_support.js]: I'm not sure I follow what you mean.  This assignment is happening just before `this._super` is called, do you want to move it afterwards or something else?

Comment (stefanpenner) [inline packages/ember-views/lib/mixins/view_support.js]: yes, as in ES6 you can't touch `this` before calling super, we should slowly move our internals to this.

Comment (rwjblue) [inline packages/ember-views/lib/mixins/view_support.js]: Updated to call `this._super` first (before using anything on `this`).
